src/commands/init.ts

```diff
@@ -0,0 +1,77 @@
+import * as path from "node:path";
+
+import enquirer from "enquirer";
+import * as fs from "fs-extra";
+
+import BaseCommand from "@/lib/base-command";
+
+export default class Init extends BaseCommand<typeof Init> {
+  protected requiresAuth = false;
+
+  static summary =
+    "Initialize a new Knock project with a knock.json configuration file.";
+
+  static description =
+    "Creates a knock.json configuration file in the current directory " +
+    "to store project-level settings like the knock resources directory.";
+
+  public async run(): Promise<void> {
+    const configPath = path.resolve(process.cwd(), "knock.json");
+
+    // 1. Check if knock.json already exists
+    const configExists = await fs.pathExists(configPath);
+    if (configExists) {
+      this.error(
+        "A knock.json file already exists in this directory. Aborting.",
+      );
+    }
+
+    // 2. Prompt user for the knock directory location
+    const { knockDir } = await enquirer.prompt<{ knockDir: string }>({
+      type: "input",
+      name: "knockDir",
+      message: "Where do you want to store your Knock resources?",
+      initial: ".knock",
+    });
+
+    // 3. Create the knock directory and resource subdirectories
+    const knockDirPath = path.resolve(process.cwd(), knockDir);
+    await fs.ensureDir(knockDirPath);
+
+    // Create resource subdirectories with .gitignore files
+    const resourceDirs = [
```

Comment:
let's specify this is an array of the `ResourceType` you specify in project-config `: ResourceType[]`